packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -372,6 +373,39 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange>
 		this.editManager.advanceMinimumSequenceNumber(brand(message.minimumSequenceNumber));
 	}
 
+	/**
+	 * Process a bunch of messages. Some important invariants for these messages:
+	 * - They are all from the same client.
+	 * - They are all part of the same batch so they are not interleaved with messages from other clients.
+	 *-  They are not interleaved with messages from other DDSes in the container.
```

Comment:
I really like the idea of calling out these invariants. Here's what comes to mind for me when I try to list everything I can think of. You're the subject matter expert on this topic, so I'll defer to you.
```suggestion
	 * Process a bunch of messages. Some important invariants for these messages:
     * - They are not interleaved with messages from other DDSes in the container.  
     * - They are all part of the same batch, which entails...  
     *   - They are contiguous in sequencing order.  
     *   - They are all from the same client.  
     *   - They are all based on the same reference sequence number.  
     *   - They are not interleaved with messages from other clients.  
```

Reply:
This list looks good. I updated it.